src/frame/mod.rs

```diff
@@ -373,6 +393,16 @@ impl Frame {
     }
 }
 
+impl PartialEq for Frame {
```

Comment:
It is because of `None` encoding - Frames should match if the encoding is None in one of them. Also without it some tests are broken (if you read Frame from file it WILL have an encoding, but if you create a new frame, it won't have encoding until you write to file (or you manually set it))

Reply:
In that case, could we also have this implementation for `Eq`? `Hash` is also a thing that still worries me, although I am not sure how to resolve that. Perhaps leaving it auto-derived is good enough

Reply:
The `Eq` doesn't do anything special, it just informs the compiler that the PartialEq impl is actually Eq. So just deriving it should be enough (source: https://doc.rust-lang.org/std/cmp/trait.Eq.html)

Reply:
Ah, good :)